services/kvstore/src/main/java/tech/pegasys/pantheon/services/kvstore/RocksDBStats.java

```diff
@@ -0,0 +1,198 @@
+package tech.pegasys.pantheon.services.kvstore;
+
+import static tech.pegasys.pantheon.metrics.MetricCategory.KVSTORE_ROCKSDB_STATS;
+
+import tech.pegasys.pantheon.metrics.prometheus.PrometheusMetricsSystem;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import io.prometheus.client.Collector;
+import org.rocksdb.HistogramData;
+import org.rocksdb.HistogramType;
+import org.rocksdb.Statistics;
+import org.rocksdb.TickerType;
+
+class RocksDBStats {
+
+  static final List<String> LABELS = Collections.singletonList("quantile");
+  static final List<String> LABEL_50 = Collections.singletonList("0.5");
+  static final List<String> LABEL_95 = Collections.singletonList("0.95");
+  static final List<String> LABEL_99 = Collections.singletonList("0.99");
+
+  // Tickers - RocksDB equivalent of counters
+  static final TickerType[] TICKERS = {
+    TickerType.BLOCK_CACHE_ADD,
+    TickerType.BLOCK_CACHE_HIT,
+    TickerType.BLOCK_CACHE_ADD_FAILURES,
+    TickerType.BLOCK_CACHE_INDEX_MISS,
+    TickerType.BLOCK_CACHE_INDEX_HIT,
+    TickerType.BLOCK_CACHE_INDEX_ADD,
+    TickerType.BLOCK_CACHE_INDEX_BYTES_INSERT,
+    TickerType.BLOCK_CACHE_INDEX_BYTES_EVICT,
+    TickerType.BLOCK_CACHE_FILTER_MISS,
+    TickerType.BLOCK_CACHE_FILTER_HIT,
+    TickerType.BLOCK_CACHE_FILTER_ADD,
+    TickerType.BLOCK_CACHE_FILTER_BYTES_INSERT,
+    TickerType.BLOCK_CACHE_FILTER_BYTES_EVICT,
+    TickerType.BLOCK_CACHE_DATA_MISS,
+    TickerType.BLOCK_CACHE_DATA_HIT,
+    TickerType.BLOCK_CACHE_DATA_ADD,
+    TickerType.BLOCK_CACHE_DATA_BYTES_INSERT,
+    TickerType.BLOCK_CACHE_BYTES_READ,
+    TickerType.BLOCK_CACHE_BYTES_WRITE,
+    TickerType.BLOOM_FILTER_USEFUL,
+    TickerType.PERSISTENT_CACHE_HIT,
+    TickerType.PERSISTENT_CACHE_MISS,
+    TickerType.SIM_BLOCK_CACHE_HIT,
+    TickerType.SIM_BLOCK_CACHE_MISS,
+    TickerType.MEMTABLE_HIT,
+    TickerType.MEMTABLE_MISS,
+    TickerType.GET_HIT_L0,
+    TickerType.GET_HIT_L1,
+    TickerType.GET_HIT_L2_AND_UP,
+    TickerType.COMPACTION_KEY_DROP_NEWER_ENTRY,
+    TickerType.COMPACTION_KEY_DROP_OBSOLETE,
+    TickerType.COMPACTION_KEY_DROP_RANGE_DEL,
+    TickerType.COMPACTION_KEY_DROP_USER,
+    TickerType.COMPACTION_RANGE_DEL_DROP_OBSOLETE,
+    TickerType.NUMBER_KEYS_WRITTEN,
+    TickerType.NUMBER_KEYS_READ,
+    TickerType.NUMBER_KEYS_UPDATED,
+    TickerType.BYTES_WRITTEN,
+    TickerType.BYTES_READ,
+    TickerType.NUMBER_DB_SEEK,
+    TickerType.NUMBER_DB_NEXT,
+    TickerType.NUMBER_DB_PREV,
+    TickerType.NUMBER_DB_SEEK_FOUND,
+    TickerType.NUMBER_DB_NEXT_FOUND,
+    TickerType.NUMBER_DB_PREV_FOUND,
+    TickerType.ITER_BYTES_READ,
+    TickerType.NO_FILE_CLOSES,
+    TickerType.NO_FILE_OPENS,
+    TickerType.NO_FILE_ERRORS,
+    // TickerType.STALL_L0_SLOWDOWN_MICROS,
+    // TickerType.STALL_MEMTABLE_COMPACTION_MICROS,
+    // TickerType.STALL_L0_NUM_FILES_MICROS,
```

Comment:
I assume we want to keep these here for visibility of all of the options?

Reply:
yes, to make it clear their absence is intentional.

Reply:
(optional) Might be worth an explanatory comment :)